bokeh/models/mappers.py

```diff
@@ -63,3 +65,229 @@ class LinearColorMapper(ColorMapper):
+    correlated to different colors.
+
+    For example, if the breaks are specified 0, 0.5, 1 and the
+    colors are defined as 'red', 'white', 'green', values 
+    between 0 and 0.5 will smoothy vary between red and white
+    while values between 0.5 and 1 will vary between white and
+    green.
+    """
+
+    palette = Seq(Color, help="""
+    A sequence of colors to use as the target palette for mapping.
+
+    This property can also be set as a ``String``, to the name of
+    any of the palettes shown in :ref:`bokeh_dot_palettes`.
+    """).accepts(Enum(Palette), lambda pal: getattr(palettes, pal))
+
+    segments = Seq(Float, help="""
+    A listing of values which are directly tied to the 
+    colors defined in the palette.  The list must have
+    at least two elements.  If there are exactly two
+    elements, then the resulting colormap will be
+    interpolated between these two values and across
+    all of the colors defined in the palette.  If there
+    are more than two colors, the length must match the
+    number of colors in the palette.  In this configuration
+    each segment value is tied to the palette color of the
+    same index.  This argument is optional.  If it is
+    not provided, the min and max of the data will be used.
+    """)
+
+    alpha = Seq(Float, help="""
+    A python array, the same length as the dictionary with the
+    desired alpha values
+    """
+    )
+
+    interpolationMethod = String(default='linear', help="""
+    A string which describes the method of interpolation.  The default, 'linear',
+    will result in a gradual transition from one color to the next.  Other
+    options are 'step' which will result in a single color for all values 
+    which fall between the segment markers (an effective floor function) 
+    """)
+
+    def __init__(self, palette=None, segments=None, alpha=1, interpolationMethod='linear', **kwargs):
+        # Elegantly handle the possability that some parameters can be passed from the
+        # kwargs structure (because of subclassing)
+        if palette is None and 'palette' in kwargs.keys():
+            palette = kwargs['palette']
+
+        if segments is None and 'segments' in kwargs.keys():
+            segments = kwargs['segments']
+
+        if alpha is None and 'alpha' in kwargs.keys():
+            alpha = kwargs['alpha']
+
+        if interpolationMethod is None and 'interpolationMethod' in kwargs.keys():
+            interpolationMethod = kwargs['interpolationMethod']
+
+
+        # Define a dict to hold any alpha values that we might extract
+        # from the different colors in the passed palette.  These can
+        # come from a 4-tuple or a
+        alpha_hold = {}
+
+        if palette is not None:
+            tmp_palette = expandPalette(palette)
+            palette = tmp_palette['palette']
+            alpha_hold = tmp_palette['alpha_hold']
+            kwargs['palette'] = palette
```

Comment:
I would be fine waiting for the serialization time.  Again, the entire idea is to offer flexibility to the end user and consistency to the JS at the same time.